File: 500-keyboard-row/keyboard-row.cpp
class Solution {
public:
    vector<string> findWords(vector<string>& words) {
        bool row1[26]={},row2[26]={},row3[26]={};

        for(char c:string("qwertyuiop")) row1[c-'a']=true;
        for(char c:string("asdfghjkl")) row2[c-'a']=true;
        for(char c:string("zxcvbnm")) row3[c-'a']=true;

        vector<string> sol;
        for(string word:words){
            bool* row;
            char first=tolower(word[0]);
            if(row1[first-'a']) row=row1;
            else if(row2[first-'a']) row=row2;
            else row=row3;

            int i=1;
            while(i<word.size()){
                if(!row[tolower(word[i])-'a']) break;
                ++i;
            }
            if(i==word.size()) sol.push_back(word);
        }
        return sol;
    }
};

File: 500-keyboard-row/keyboard-row.py
class Solution:
    def findWords(self, words: List[str]) -> List[str]:
        row = [0 for _ in range(26)]
        sol = []

        for i in range(26):
            char = chr(i+ord("a"))
            if char in "qwertyuiop":
                row[i]=1
            elif char in "asdfghjkl":
                row[i]=2
            else: row[i]=3

        for word in words:
            chr_idx = ord(word[0].lower())-ord("a")
            row_pos = row[chr_idx]
            isPossible = True

            for char in word:
                char_row_pos = row[(ord(char.lower())-ord("a"))]
                if char_row_pos!=row_pos:
                    isPossible = False
                    break
            if isPossible:
                sol.append(word)
        
        return sol